open_spiel/spiel.h

```diff
@@ -757,6 +757,34 @@ class Game : public std::enable_shared_from_this<Game> {
   // of chance nodes are not included in this length.
   virtual int MaxGameLength() const = 0;
 
+  // The maximum number of chance nodes occurring in any history of the game.
+  // This is typically something like the number of times dice are rolled.
+  virtual int MaxChanceNodesInHistory() const {
+    // This function is not a pure virtual like MaxGameLength()
+    // for compatibility reasons.
+    SpielFatalError("MaxChanceNodesInHistory() is not implemented");
+  }
+
+  // The maximum number of moves in the game. The value State::MoveNumber()
+  // must never be higher than this value.
+  virtual int MaxMoveNumber() const {
+    return MaxGameLength() + MaxChanceNodesInHistory();
+  }
+
+  // The maximum length of any history in the game.
+  // The value State::History().size() must never be higher than this value.
+  virtual int MaxHistoryLength() const {
+    if (GetType().dynamics == GameType::Dynamics::kSimultaneous) {
```

Comment:
Doesn't this belong in SimMoveGame? It's a particular implementation choice.

Reply:
I think currently sim move games are not required to derive from SimMoveGame (see goofspiel). This is more generic implementation. I'd leave it here in one place rather than in many places over the codebase.

Reply:
I'm ok with this. @elkhrt , does it work for you?

Reply:
Would like to include this in tomorrow's update, if possible.